statsforecast/core.py

```
@@ -838,18 +536,39 @@ def _validate_model_names(self):
                 "Model names must be unique. You can use `alias` to set a unique name for each model."
             )
 
-    def _prepare_fit(self, df, sort_df):
-        if df is not None:
-            df_process = DataFrameProcessing(df, sort_df)
-            self.ga = df_process.grouped_array()
-            self.uids = df_process.indices
-            self.last_dates = df_process.dates
-            self.ds = df_process.index
-            self.og_dates = df_process.np_df["ds"]
-            self.og_unique_id = df_process.np_df["unique_id"]
-            self.engine = df_process.engine_dataframe
-            self.n_jobs = _get_n_jobs(len(self.ga), self.n_jobs)
-            self.sort_df = sort_df
+    def _prepare_fit(
+        self, df: Optional[DataFrame], sort_df: bool = True, save_original: bool = False
+    ) -> None:
+        if df is None:
+            _warn_df_constructor()
+            return
+        df = ensure_time_dtype(df, "ds")
+        if isinstance(df, pd.DataFrame):
+            if df.index.name == "unique_id":
+                warnings.warn(
+                    "Passing unique_id as the index is deprecated. "
```

If it's deprecated, should we raise an exception instead?